cvat/apps/quality_control/quality_reports.py

```diff
@@ -2082,19 +2108,52 @@ def generate_report(self) -> ComparisonReport:
             ),
         )
         mean_ious = []
+        empty_frame_count = 0
         confusion_matrix_labels, confusion_matrix, _ = self._make_zero_confusion_matrix()
 
-        for frame_id, frame_result in self._frame_results.items():
-            intersection_frames.append(frame_id)
-            conflicts += frame_result.conflicts
+        for frame_result in frame_summaries.values():
             confusion_matrix += frame_result.annotations.confusion_matrix.rows
 
+            if not np.any(frame_result.annotations.confusion_matrix.rows):
+                empty_frame_count += 1
+
             if annotation_components is None:
                 annotation_components = deepcopy(frame_result.annotation_components)
             else:
                 annotation_components.accumulate(frame_result.annotation_components)
+
             mean_ious.append(frame_result.annotation_components.shape.mean_iou)
 
+        annotation_summary = self._compute_annotation_summary(
+            confusion_matrix, confusion_matrix_labels
+        )
+
+        if self.settings.match_empty_frames and empty_frame_count:
+            # Add virtual annotations for empty frames,
+            # they are not included in the confusion matrix
+            annotation_summary.valid_count += empty_frame_count
+            annotation_summary.total_count += empty_frame_count
+            annotation_summary.ds_count += empty_frame_count
+            annotation_summary.gt_count += empty_frame_count
```

Comment:
I'm not sure if this is really a useful thing to do here. Imagine a situation where there are 100 frames, frame 1 has 1 valid annotations and 2 total annotations, every other frame is empty.

In this case, with `match_empty_frames` off:

* frame 1 has accuracy 50%, others have 0%.
* total accuracy is 50%.

With `match_empty_frames` on:

* frame 1 has accuracy 50%, others have 100%.
* total accuracy is (1+99)/(2+99) = 99%.

Do you think this jump in total accuracy would be expected by the user?

Reply:
It's hard to say about the expectations, to me the user just works in one of the modes - they either consider empty frames as annotated or not. But the second value seems to be more correct, so maybe something should be done for the case with the option disabled.